File: alurapic/src/app/errors/global-error-handler/log.ts
export interface ILog {
    id: number;
    user: string;
    message: string;
    url: string;
    stack: string;
}

export class LogCreate {
    message: string;
    url: string;
    stack: string;

    constructor(message: string, url: string, stack: string) {
        this.message = message;
        this.url = url;
        this.stack = stack;
    }
}

File: api/Models/Log.cs
using System;

namespace api.Models
{
    public class Log
    {
        private static int _currentId = 0;

        public Log(string message, string user, string url, string stack) 
        {
            Id = ++_currentId;
            Message = message;
            User = user;
            Url = string.IsNullOrWhiteSpace(url) ? null : url;
            Stack = stack;
        }

        public int Id {get;set;}
        public string Message {get;set;}
        public string User {get;set;}
        public string? Url {get;set;}
        public string Stack {get;set;}
    }
}